packages/api/src/utils/client/httpClient.ts

```diff
@@ -279,6 +286,14 @@ export class HttpClient implements IHttpClient {
       if (bearerToken && headers["Authorization"] === undefined) {
         headers["Authorization"] = `Bearer ${bearerToken}`;
       }
+      if (url.username || url.password) {
+        if (headers["Authorization"] === undefined) {
+          headers["Authorization"] = `Basic ${toBase64(`${url.username}:${url.password}`)}`;
```

Comment:
If we can move this logic to a utility function `generateAuthHeader(username, password)`, that will be easy to manage in future.

Reply:
Was consider that as well but opted against it as same argument could be made for setting bearer token above.

Reason why I opted against it
- not reused anywhere else
- simple enough
- format of auth header as defined in RFC will never change